### PR TITLE
Storage Documentation Update

### PR DESCRIPTION
**Pull request by marcbachmann**

### CODE REVIEW

File: reference-docs/server-configuration/config.md [hunk 1]
```diff
@@ -127,11 +131,9 @@ auth: {
...
-        secret: '8wP0oTmQFX4^f@FCW*MiQ%qkDu*m6B!NBElVg%hVmMHNdr2F@F9J4yf$ykN5',
-        bcrypt_iterations: 10,
-        password_reset_url: '/reset#code={{code}}',
-        denied_passwords: [
+        defaultLoginDomain: 'livingdocs.io',
+        passwordResetUrl: '/reset#code={{code}}',
+        deniedPasswords: [
           'upfront',
           'livingd[o0][cg]s',
           'lukas',
```
Comment (DaRaFF) [inline reference-docs/server-configuration/config.md:139]: :-)

File: reference-docs/server-configuration/config.md [hunk 2]
```diff
@@ -339,13 +351,13 @@ images: {
...
-  upload_restrictions: {
-    max_file_size: 100*1000*1000 // 100MB, defaults to 5MB.
+  uploadRestrictions: {
+    maxFileSize: 100*1000*1000 // 100MB, defaults to 5MB.
   }
 }
 ```
 
-- Convert Option
+#### images.processing.convert Confguration
```
Comment (DaRaFF) [inline reference-docs/server-configuration/config.md:360]: Confguration -> Configuration

File: reference-docs/server-configuration/config.md [hunk 3]
```diff
@@ -357,19 +369,20 @@ processing: {
 
 #### Videos
 Added in: `release-2021-02`
-For videos it is necessary to have a specific configuration. The storage, publicUrl and the uploadRestriction must be set.
+For videos it is necessary to have a specific configuration.
+The storage, publicUrl and the uploadRestriction must be set.
+
+Consult the [storage configuration](/reference/storage-strategy-configuration.md) to configure other providers than s3.
```
Comment (DaRaFF) [inline reference-docs/server-configuration/config.md:375]: 👍

File: reference-docs/server-configuration/config.md [hunk 4]
```diff
@@ -382,30 +395,32 @@ For videos it is necessary to have a specific configuration. The storage, public
     }
   }
 ```
-[config options](/reference/config-storage-options.md) for `storage`.
 
 #### Files
 
+Consult the [storage configuration](/reference/storage-strategy-configuration.md) to configure other providers than s3.
```
Comment (DaRaFF) [inline reference-docs/server-configuration/config.md:401]: You could add a comment why one would need this files config.

File: reference-docs/server-configuration/config.md [hunk 5]
```diff
@@ -533,15 +548,15 @@ Configure the elasticsearch instance used by the search feature.
 ```js
 search: {
   host: 'http://localhost:9200',
-  article_document_index: 'li_local_documents',
-  number_of_replicas: 1,
+  articleDocumentIndex: 'li_local_documents',
```
Comment (DaRaFF) [inline reference-docs/server-configuration/config.md:551]: _ are not supported anymore in the future. You could already update the example.

File: reference-docs/server-configuration/logging.md [hunk 6]
```diff
@@ -2,34 +2,37 @@
...
+
+If Errors are passed as an attribute of an object,
+the error will just be coerced as string with `toString()` and not serialized with its stack trace.
 
-If Errors are passed as a property of an object, e.g. log.error({x: new Error()}),
-the error will just be coerced as string with `toString()` and not serialized
-with its stack trace.
+Make sure you always pass it directly or use `err` or `error` attributes. That behaviour is defined by [`serializers`](https://getpino.io/#/docs/api?id=serializers-object)
 
+- ❌ `log.error({x: new Error()})`
```
Comment (DaRaFF) [inline reference-docs/server-configuration/logging.md:29]: 👍

File: reference/storage-strategy-configuration.md [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+```js
+storage: {
+  strategy: 'fs',
+  config: {
+    path: require('os').tmpdir()
+  }
+}
+```
+
+### S3 Object Storage
```
Comment (DaRaFF) [inline reference/storage-strategy-configuration.md:52]: I would put this to the first place when it's the most common one.

File: reference/storage-strategy-configuration.md [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+  }
+}
+```
+
+#### S3 with a HTTP Proxy
+
+If you use an http proxy like [squid](http://squid-cache.org/), you can declare the `HTTPS_PROXY` environment variable, which then configures the aws client.
+e.g. `HTTPS_PROXY=http://localhost:3128 node index.js`
+
+Or you cna explicitly configure it with the options provided by the aws sdk:
```
Comment (DaRaFF) [inline reference/storage-strategy-configuration.md:77]: cna -> can

File: reference/storage-strategy-configuration.md [hunk 9]
```diff
@@ -0,0 +1,136 @@
...
+      token_uri: 'https://oauth2.googleapis.com/token',
+      auth_provider_x509_cert_url: 'https://www.googleapis.com/oauth2/v1/certs',
+      client_x509_cert_url: '******'
+    }
+  }
+}
+```
+
+### Cloudinary
+
```
Comment (DaRaFF) [inline reference/storage-strategy-configuration.md:118]: Added in: `release-2020-12`

**Review by DaRaFF — APPROVED** [review]
Awesome, a real improvement 👍

File: reference/storage-strategy-configuration.md (hunk 7 above)
Comment (marcbachmann) [reply reference/storage-strategy-configuration.md:52]: I'd like to have the short and simple ones first. It's easier to scan through and to get the concept.

File: reference-docs/server-configuration/config.md (hunk 1 above)
Comment (marcbachmann) [reply reference-docs/server-configuration/config.md:139]: that won't be removed. ever 😆

**Pull request merged**